internal/controller/runtime/fsm/runtime_fsm_select_shoot_processing.go

```diff
@@ -19,38 +19,42 @@ func sFnSelectShootProcessing(_ context.Context, m *fsm, s *systemState) (stateF
 	if s.shoot.Spec.DNS == nil || s.shoot.Spec.DNS.Domain == nil {
 		msg := fmt.Sprintf("DNS Domain is not set yet for shoot: %s, scheduling for retry", s.shoot.Name)
 		m.log.Info(msg)
-		return updateStatusAndRequeueAfter(m.RCCfg.GardenerRequeueDuration)
+		return requeueAfter(m.RCCfg.GardenerRequeueDuration)
 	}
 
 	lastOperation := s.shoot.Status.LastOperation
 	if lastOperation == nil {
 		msg := fmt.Sprintf("Last operation is nil for shoot: %s, scheduling for retry", s.shoot.Name)
 		m.log.Info(msg)
-		return updateStatusAndRequeueAfter(m.RCCfg.GardenerRequeueDuration)
+		return requeueAfter(m.RCCfg.GardenerRequeueDuration)
 	}
 
 	patchShoot, err := shouldPatchShoot(&s.instance, s.shoot)
 	if err != nil {
 		msg := fmt.Sprintf("Failed to get applied generation for shoot: %s, scheduling for retry", s.shoot.Name)
 		m.log.Error(err, msg)
-		return updateStatusAndStop()
+		return requeueAfter(m.RCCfg.GardenerRequeueDuration)
 	}
 
 	if patchShoot {
 		m.log.Info("Gardener shoot already exists, updating")
 		return switchState(sFnPatchExistingShoot)
 	}
 
-	if lastOperation.Type == gardener.LastOperationTypeCreate {
-		return switchState(sFnWaitForShootCreation)
-	}
+	if s.instance.Status.State == imv1.RuntimeStatePending || s.instance.Status.State == "" {
+		if lastOperation.Type == gardener.LastOperationTypeCreate {
+			return switchState(sFnWaitForShootCreation)
+		}
 
-	if lastOperation.Type == gardener.LastOperationTypeReconcile {
-		return switchState(sFnWaitForShootReconcile)
+		if lastOperation.Type == gardener.LastOperationTypeReconcile {
+			return switchState(sFnWaitForShootReconcile)
+		}
 	}
 
-	m.log.Info("Unknown shoot operation type, exiting with no retry")
-	return stopWithMetrics()
+	// All other runtimes in Ready and Failed state will be not processed to mitigate massive reconciliation during restart
+	msg := fmt.Sprintf("Stopping processing reconcile for runtime %s and shoot %s, exiting with no retry:", s.instance.Name, s.shoot.Name)
+	m.log.Info(msg)
+	return stop()
```

Comment:
why have you decided to call `stop()` here instead of `stopWithMetrics()`?

Reply:
stopWithMetrics() original usage was to record unusual situation, shat should concern developers. After my change this stop() is is actually dropping all runtimes wich were not classified as candidates for patch or to continue any pending operation. In case of our load tests that would be all 1000/1000 runtimes in case of the restart of the pod when all runtimes are ready. That makes here this metrics unnecessary